google-cloud-core/src/main/java/com/google/cloud/CivilDateTime.java

```diff
@@ -0,0 +1,134 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+import com.google.api.client.util.Lists;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a date and time with no timezone, e.g. e.g. 2017-03-17 07:41:35.220000.
+ *
+ * CivilDateTime holds values for BigQuery DATETIME
+ * (https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime-type).
+ */
+public class CivilDateTime implements Comparable<CivilDateTime> {
+
+  private final CivilDate date;
+  private final CivilTime time;
+
+  private CivilDateTime(CivilDate date, CivilTime time) {
+    Preconditions.checkNotNull(date, "Date cannot be null");
+    Preconditions.checkNotNull(time, "Time cannot be null");
+    this.date = date;
+    this.time = time;
+  }
+
+  /**
+   * Constructs a new CivilDate instance with time set to the beginning of the day.
+   *
+   * @param date civil date
+   */
+  public static CivilDateTime ofDate(CivilDate date) {
+    return ofDateAndTime(date, CivilTime.ofHoursMinutesSecondsNanos(0, 0, 0, 0));
+  }
+
+  /**
+   * Constructs a new CivilDate instance.
+   *
+   * @param date civil date
+   * @param time civil time
+   */
+  public static CivilDateTime ofDateAndTime(CivilDate date, CivilTime time) {
+    return new CivilDateTime(date, time);
+  }
+
+  /**
+   * @param dateTime Date plus time: (yyyy-MM-dd HH:mm:ss.SSSSSSSSS) or
+   * (yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSS).
+   */
+  public static CivilDateTime parseDateTime(String dateTime) {
+    String splitToken = " ";
+    if (dateTime.contains("T")) {
+      splitToken = "T";
+    }
+    List<String> parts = Lists.newArrayList(Splitter.on(splitToken).split(dateTime));
+    if (parts.isEmpty()) {
+      throw new IllegalArgumentException("Invalid date time format: " + dateTime);
+    }
+    CivilDate date = CivilDate.parseDate(parts.get(0));
+    if (parts.size() == 1) {
+      return ofDate(date);
+    } else if (parts.size() == 2) {
+      return ofDateAndTime(date, CivilTime.parseTime(parts.get(1)));
+    } else {
+      throw new IllegalArgumentException("Invalid date time format: " + dateTime);
+    }
+  }
+
+  /** Returns the date component of this date-time. */
+  public CivilDate getDate() {
+    return date;
+  }
+
+  /** Returns the time component of this date-time. */
+  public CivilTime getTime() {
+    return time;
+  }
+
+  /** Formats the date and time in the format yyyy-MM-dd HH:mm:ss.SSSSSSSSS */
+  public String formatMicroCloudTime() {
```

Comment:
My thought was that since it uses a space instead of a 'T', that should be distinguished somehow, thus the "Cloud" part.